Prelude/NonEmpty.swift

```diff
@@ -0,0 +1,96 @@
+public struct NonEmpty<Collection: Swift.Collection> {
+  public typealias Element = Collection.Iterator.Element
+  public let head: Collection.Iterator.Element
+  public let tail: Collection
+}
+
+infix operator >|: AdditionPrecedence
```

Comment:
Considered putting this in `Operators.swift`, but wondering if less generalized ones like this (and the lens-specific ones) ought to be in the file that uses them.